addons/l10n_th/views/res_company_view.xml

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <data>
+        <record id="view_company_form_inherit_l10n_th" model="ir.ui.view">
+            <field name="name">res.company.form.inherit.l10n_th</field>
+            <field name="model">res.company</field>
+            <field name="inherit_id" ref="base.view_company_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//field[@name='company_registry']" position="attributes">
+                    <attribute name="invisible" add="country_code == 'TH'" separator=" or "/>
+                </xpath>
```

Comment:
*applies to all xpaths*

XPath is good for more complex inheritance, but in this case, yours is simple.
So it's easier if you write:


```suggestion
                <field name="company_registry" position="attributes">
                    <attribute name="invisible" add="country_code == 'TH'" separator=" or "/>
                </field>
```